go-controller/pkg/ovn/controller/admin_network_policy/metrics.go

```diff
@@ -0,0 +1,130 @@
+	prometheus.MustRegister(c)
+}
+
+func (c *Controller) teardownMetricsCollector() {
+	prometheus.Unregister(c)
+}
+
+// Describe is implemented with DescribeByCollect. That's possible because the
+// Collect method will always return the same two metrics with the same two
+// descriptors.
+func (c *Controller) Describe(ch chan<- *prometheus.Desc) {
+	prometheus.DescribeByCollect(c, ch)
+}
+
+func (c *Controller) fetchANPRuleCountMetric(isBanp bool) map[string]map[string]int {
+	c.RLock()
+	defer c.RUnlock()
+	ruleCount := map[string]map[string]int{
+		string(libovsdbutil.ACLIngress): {
+			string(anpapi.AdminNetworkPolicyRuleActionAllow): 0,
+			string(anpapi.AdminNetworkPolicyRuleActionDeny):  0,
+		},
+		string(libovsdbutil.ACLEgress): {
+			string(anpapi.AdminNetworkPolicyRuleActionAllow): 0,
+			string(anpapi.AdminNetworkPolicyRuleActionDeny):  0,
+		},
+	}
+	if !isBanp {
+		// its ANP; so add Pass label
+		ruleCount[string(libovsdbutil.ACLIngress)][string(anpapi.AdminNetworkPolicyRuleActionPass)] = 0
+		ruleCount[string(libovsdbutil.ACLEgress)][string(anpapi.AdminNetworkPolicyRuleActionPass)] = 0
+	}
+	for _, state := range c.anpCache {
+		allow, pass, deny := c.countRules(state.ingressRules)
+		ruleCount[string(libovsdbutil.ACLIngress)][string(anpapi.AdminNetworkPolicyRuleActionAllow)] += allow
+		ruleCount[string(libovsdbutil.ACLIngress)][string(anpapi.AdminNetworkPolicyRuleActionDeny)] += deny
+		if !isBanp {
+			ruleCount[string(libovsdbutil.ACLIngress)][string(anpapi.AdminNetworkPolicyRuleActionPass)] += pass
+		}
+		allow, pass, deny = c.countRules(state.egressRules)
+		ruleCount[string(libovsdbutil.ACLEgress)][string(anpapi.AdminNetworkPolicyRuleActionAllow)] += allow
+		ruleCount[string(libovsdbutil.ACLEgress)][string(anpapi.AdminNetworkPolicyRuleActionDeny)] += deny
+		if !isBanp {
+			ruleCount[string(libovsdbutil.ACLEgress)][string(anpapi.AdminNetworkPolicyRuleActionPass)] += pass
+		}
+	}
+	return ruleCount
+}
+
+func (c *Controller) countRules(rules []*gressRule) (int, int, int) {
+	var passCount, allowCount, denyCount int
+	for _, rule := range rules {
+		switch rule.action {
+		case nbdb.ACLActionAllowRelated:
+			allowCount++
+		case nbdb.ACLActionDrop:
+			denyCount++
+		case nbdb.ACLActionPass:
+			passCount++
+		default:
+			panic(fmt.Sprintf("Failed to count rule type: unknown acl action %s", rule.action))
+		}
+	}
+	return allowCount, passCount, denyCount
+}
+
+// Collect first triggers the fetchANPRuleCountMetric. Then it
+// creates constant metrics for each host on the fly based on the returned data.
+//
+// Note that Collect could be called concurrently, so we depend on
+// fetchANPRuleCountMetric to be concurrency-safe.
+func (c *Controller) Collect(ch chan<- prometheus.Metric) {
+	aRuleCount := c.fetchANPRuleCountMetric(false)
+	for direction, actions := range aRuleCount {
+		for action, count := range actions {
+			ch <- prometheus.MustNewConstMetric(
+				anpRuleCountDesc,
+				prometheus.CounterValue,
```

Comment:
I think rules count is a gauge and not counter? ref https://prometheus.io/docs/concepts/metric_types/

Reply:
oh good catch Nadia!